File: 217 Contains Duplicate/217 Contains Duplicate.swift
class Solution {
  func containsDuplicate(_ nums: [Int]) -> Bool {
    var numsSet: Set<Int> = []

    for num in nums {
      if numsSet.contains(num) {
        return true
      }

      numsSet.insert(num)
    }

    return false
  }
}

File: 217 Contains Duplicate/217 Contains Duplicate.ts
function containsDuplicate(nums: number[]): boolean {
  const numsSet: Set<number> = new Set();

  for (const num of nums) {
    if (numsSet.has(num)) {
      return true;
    }

    numsSet.add(num);
  }

  return false;
}
